src/krux/pages/file_manager.py

```diff
@@ -78,8 +78,11 @@ def select_file(
 
                 del dir_files
 
-                # show sorted folders first than sorted files
-                for filename in directories + files:
+                # show sorted folders first then sorted files
+                for filename, is_directory in [(x, True) for x in directories] + [
```

Comment:
I'm understanding this as a memory efficiency difference:

While iterating over `[(x, True) for x in directories] + [(x, False) for x in files]:`, that expression will create a new list in memory as big as the other 2 existing lists (2x more memory than needed).

Whereas enumerate() will return an efficient iterator/generator that step through the 2 existing lists without needing to create a new one.  The extra step of setting is_directory is negligible because len(a_list) is not actually computing the number of elements in the list, it's returning the internal counter efficiently.